jsapp/js/account/plan.component.tsx

```diff
@@ -0,0 +1,403 @@
+            listItems.push({icon: iconBool, item: itemName});
+          }
+        }
+      })
+    );
+    return listItems;
+  };
+
+  const checkMetaFeatures = () => {
+    let expandBool = false;
+    if (state.products.length >= 0) {
+      filterPrices().map((price) => {
+        for (const featureItem in price.metadata) {
+          if (
+            featureItem.includes('feature_support_') ||
+            featureItem.includes('feature_advanced_') ||
+            featureItem.includes('feature_addon_')
+          ) {
+            expandBool = true;
+            break;
+          }
+        }
+      });
+    }
+    return expandBool;
+  };
+
+  const returnListItem = (type: string, name: string): ReactNode => {
+    return getListItem(type, name).map((listItem) =>
+      listItem.icon ? (
+        listItem.icon === true && (
+          <li key={listItem.item}>
+            <div className={styles.iconContainer}>
+              <Icon
+                name='check'
+                size='m'
+                classNames={
+                  name === 'Professional plan'
+                    ? [styles.tealCheck]
+                    : [styles.stormCheck]
+                }
+              />
+            </div>
+            {listItem.item}
+          </li>
+        )
+      ) : (
+        <li key={listItem.item}>
+          <div className={styles.iconContainer}>
+            <Icon name='close' size='m' classNames={[styles.redClose]} />
+          </div>
+          {listItem.item}
+        </li>
+      )
+    );
+  };
+
+  if (!state.products.length) {
+    return null;
+  }
+  return (
+    <div className={styles.accountPlan}>
+      <div className={styles.plansSection}>
+        <form className={styles.intervalToggle}>
+          <input
+            type='radio'
+            id='switch_left'
+            name='switchToggle'
+            value='year'
+            onChange={() => dispatch({type: 'year'})}
+            checked={!state.filterToggle}
+            aria-label={'Toggle to annual options'}
+          />
+          <label htmlFor='switch_left'>{t('Annual')}</label>
+
+          <input
+            type='radio'
+            id='switch_right'
+            name='switchToggle'
+            value='month'
+            aria-label={'Toggle to month options'}
+            onChange={() => dispatch({type: 'month'})}
+            checked={state.filterToggle}
+          />
+          <label htmlFor='switch_right'> {t('Monthly')}</label>
+        </form>
+        <div
+          className={styles.currentPlan}
+          style={{
+            gridRow: 0,
+            gridColumn: 1 + filterPrices().findIndex(isSubscribedProduct),
+            display:
+              filterPrices().findIndex(isSubscribedProduct) >= 0 ? '' : 'none',
+          }}
+        >
+          {t('your plan')}
+        </div>
+        {filterPrices().map((price: Price, i: number) => (
+          <div className={styles.planContainer} key={i}>
```

Comment:
React gets kind of iffy when using an index as a key - see [here](https://react.dev/learn/rendering-lists), under 'Pitfall'. This should get replaced with `key={price.id}` or some other unique value.